lib/cosmos/gui/choosers/integer_chooser.rb

```diff
@@ -12,56 +12,38 @@
 require 'cosmos/gui/choosers/value_chooser'
 
 module Cosmos
-
-  class IntegerChooserIntValidator < Qt::IntValidator
-    def initialize(*args)
-      super(*args)
-    end
-
-    def fixup(input)
-      begin
-        value = input.to_i
-        if value < bottom()
-          # Handle less than bottom
-          parent().setText(bottom().to_s)
-        elsif value > top()
-          # Handle greater than top
-          parent().setText(top().to_s)
-        elsif input != value.to_s
-          # Handle poorly formatted (only known case is float given as starting value)
-          parent().setText(value.to_s)
-        end
-      rescue Exception => err
-        # Oh well no fixup
-      end
-    end
-  end
-
+  # Widget which creates a horizontally laid out label and editable integer value.
+  # Minimum and maximum values can be specified to perform input validation.
+  # A callback can be specified which is called once the value is changed.
   class IntegerChooser < ValueChooser
-    # Callback called when the value changes
-    attr_accessor :sel_command_callback
-
+    # @param parent (see ValueChooser#initialize)
+    # @param label_text (see ValueChooser#initialize)
+    # @param initial_value (see ValueChooser#initialize)
+    # @param minimum_value [Integer] Minimum allowable value
+    # @param maximum_value [Integer] Maximum allowable value
+    # @param field_width (see ValueChooser#initialize)
+    # @param fill (see ValueChooser#initialize)
     def initialize(parent, label_text, initial_value,
                    minimum_value = nil, maximum_value = nil,
                    field_width = 20, fill = false)
       super(parent, label_text, initial_value, field_width, fill)
       @minimum_value = minimum_value
       @maximum_value = maximum_value
 
-      validator = IntegerChooserIntValidator.new(@value)
+      validator = Qt::IntValidator.new(@value)
```

Comment:
The one I wrote contains a modified fixup method... Are you sure the existing one works the same?